### PR TITLE
Update helm chart for running as deployment

### PR DESCRIPTION
**Pull request by praxist**
Builds off of https://github.com/kubernetes-sigs/descheduler/pull/572 to also add the deployment to helm.

`helm template` yields the following with `kind: Deployment`:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: RELEASE-NAME-descheduler
  labels:
    app.kubernetes.io/name: descheduler
    helm.sh/chart: descheduler-0.20.0
    app.kubernetes.io/instance: RELEASE-NAME
    app.kubernetes.io/version: "0.20.0"
    app.kubernetes.io/managed-by: Tiller
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: descheduler
      app.kubernetes.io/instance: RELEASE-NAME
  template:
    metadata:
      labels:
        app.kubernetes.io/name: descheduler
        app.kubernetes.io/instance: RELEASE-NAME
      annotations:
        checksum/config: 9fe081cdd7d5c040105f0b817cfded705b3a18c3e4258d0afef86ec00f1cf9ee
    spec:
      priorityClassName: system-cluster-critical
      serviceAccountName: RELEASE-NAME-descheduler
      containers:
        - name: descheduler
          image: "k8s.gcr.io/descheduler/descheduler:v0.20.0"
          imagePullPolicy: IfNotPresent
          command:
            - "/bin/descheduler"
          args:
            - "--policy-config-file"
            - "/policy-dir/policy.yaml"
            - "--descheduling-interval"
            - 5m
            - "--v"
            - "3"
          ports:
            - containerPort: 10258
              protocol: TCP
          resources:
            requests:
              cpu: 500m
              memory: 256Mi

          securityContext:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - ALL
            privileged: false
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            # Required until 0.21.0 image is released
            # https://github.com/kubernetes-sigs/descheduler/issues/466
            runAsUser: 1000
          volumeMounts:
            - mountPath: /policy-dir
              name: policy-volume
      volumes:
        - name: policy-volume
          configMap:
            name: RELEASE-NAME-descheduler
```

### CODE REVIEW

**Comment by praxist** [comment]
I signed it

**Review by damemi — COMMENTED** [review]
/ok-to-test

**Comment by damemi** [comment]
Sorry, this will need a rebase

**Comment by praxist** [comment]
@damemi should be ready for re-review now

File: charts/descheduler/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    metadata:
+      labels:
+        {{- include "descheduler.selectorLabels" . | nindent 8 }}
+        {{- if .Values.podLabels }}
+        {{- .Values.podLabels | toYaml | nindent 8 }}
+        {{- end }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- if .Values.podAnnotations }}
+        {{- .Values.podAnnotations | toYaml | nindent 12 }}
```
Comment (cnmcavoy) [inline charts/descheduler/templates/deployment.yaml:23]: I think this is supposed to be `nindent 8`. (see the indent level for labels above)

File: charts/descheduler/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            - mountPath: /policy-dir
+              name: policy-volume
+      volumes:
+        - name: policy-volume
+          configMap:
+            name: {{ template "descheduler.fullname" . }}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
```
Comment (a7i) [inline charts/descheduler/templates/deployment.yaml:70]: Would be great to include tolerations similar to the one in CronJob https://github.com/kubernetes-sigs/descheduler/pull/562

Comment (praxist) [reply charts/descheduler/templates/deployment.yaml:70]: added

File: charts/descheduler/templates/deployment.yaml (hunk 1 above)
Comment (praxist) [reply charts/descheduler/templates/deployment.yaml]: good catch - done

**Comment by damemi** [comment]
/hold
@praxist could you squash to 1 commit please?

**Comment by praxist** [comment]
@damemi rebased + squashed

**Review by damemi — COMMENTED** [review]
/hold cancel
/lgtm

**Pull request merged**